client/my-sites/site-settings/form-general.jsx

```diff
@@ -414,7 +415,7 @@ export class SiteSettingsFormGeneral extends Component {
 							) }
 						</>
 					) }
-				{ ! isNonAtomicJetpackSite && (
+				{ ! isNonAtomicJetpackSite && ! isWpcomStagingSite && (
```

Comment:
Could we re-use the same `FormLabel`/`FormRadio` independently if `isWpcomStagingSite`?

Something like this:
```suggestion
				{ ! isNonAtomicJetpackSite && (
```

```diff
					<FormLabel className="site-settings__visibility-label is-public">
						<FormRadio
							name="blog_public"
							value="1"
							checked={
								( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
								( blogPublic === 0 && ! wpcomPublicComingSoon ) ||
								blogPublic === 1
							}
							onChange={ () =>
								this.handleVisibilityOptionChange( {
-									blog_public: 1,
+									blog_public: isWpcomStagingSite ? 0 : 1,
									wpcom_coming_soon: 0,
									wpcom_public_coming_soon: 0,
								} )
							}
							disabled={ isRequestingSettings }
							onClick={ eventTracker( 'Clicked Site Visibility Radio Button' ) }
							label={ translate( 'Public' ) }
						/>
					</FormLabel>
```